temporian/core/operators/test/test_tick_calendar.py

```diff
@@ -25,60 +25,114 @@ class TickCalendarOperatorTest(parameterized.TestCase):
         },
         {
             # Test: mday=31, February should be ignored
             "span": ["2020-01-01 00:00", "2020-03-31 00:00"],
             "calendar_args": dict(mday=31),
-            "expected": [
+            "left": "2019-12-31 00:00",
+            "ticks": [
                 "2020-01-31 00:00",
                 "2020-03-31 00:00",
             ],
+            "right": None,
         },
         {
             # Test: offsets in hours at beginning and end
             "span": ["2020-01-01 13:04", "2020-03-06 19:35"],
             "calendar_args": dict(mday=1),
-            "expected": ["2020-02-01 00:00", "2020-03-01 00:00"],
+            "left": "2020-01-01 00:00",
+            "ticks": [
+                "2020-02-01 00:00",
+                "2020-03-01 00:00",
+            ],
+            "right": "2020-04-01 00:00",
         },
         {
             # Test: 3 days in the span, should only consider start/end
             "span": ["2020-01-01", "2020-01-02", "2020-01-03"],
             "calendar_args": dict(mday="*"),
-            "expected": ["2020-01-01", "2020-01-02", "2020-01-03"],
+            "left": None,
+            "ticks": ["2020-01-01", "2020-01-02", "2020-01-03"],
+            "right": None,
         },
         {
             # Test: specific mday/hour and offsets
             "span": ["2020-01-02 13:04", "2020-03-05 19:35"],
             "calendar_args": dict(hour=3, mday=5),
-            "expected": ["2020-02-05 03:00", "2020-03-05 03:00"],
+            "left": "2019-12-05 03:00",
+            "ticks": [
+                "2020-01-05 03:00",
+                "2020-02-05 03:00",
+                "2020-03-05 03:00",
+            ],
+            "right": "2020-04-05 03:00",
         },
         {
             # Test: Edge cases for range
             "span": ["2020-01-10 13:00", "2020-03-10 12:59:59.99"],
             "calendar_args": dict(hour=13, mday=10),
-            "expected": ["2020-01-10 13:00", "2020-02-10 13:00"],
+            "left": None,
+            "ticks": [
+                "2020-01-10 13:00",
+                "2020-02-10 13:00",
+            ],
+            "right": "2020-03-10 13:00",
+        },
+        {
+            # Test: Edge cases for range
+            "span": ["2020-01-10 12:59:59.99", "2020-03-10 13:00"],
+            "calendar_args": dict(hour=13, mday=10),
+            "left": "2019-12-10 13:00",
+            "ticks": [
+                "2020-01-10 13:00",
+                "2020-02-10 13:00",
+                "2020-03-10 13:00",
+            ],
+            "right": None,
         },
     )
-    def test_base(self, span, expected, calendar_args):
-        evset = event_set(timestamps=span)
-        expected_evset = event_set(timestamps=expected)
-        result = evset.tick_calendar(**calendar_args)
+    def test_base(self, span, left, ticks, right, calendar_args):
+        for include_right in [None, True, False]:
+            for include_left in [None, True, False]:
```

Comment:
What about a parameterized test?

Reply:
this one is a bit tricky, I have the test cases defined and want to try all the combinations of left/right inclusion. I could define the test cases outside and programmatically expand them to all the combinations before calling `@parameterized.parameters`